### PR TITLE
refactor: Client options now have `wallet` field to pre-configure wallet

### PR DESCRIPTION
**Pull request by jawid-h**
### Issue being fixed or implemented  

Client options now accept `wallet` parameter of type `Wallet.Options` to introduce a namespace for wallet options. It should now be easier to configure wallet with `privateKey` instead of a `mnemonic` for example or do not initialize it if `wallet` param is not passed.

### What was done  

- Updated `Client` class constructor and `ClientOpts`

### How Has This Been Tested?

Existing tests have been updated and re-runned.

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

**Comment by Alex-Werner** [comment]
Very nice ! 
Thank you ! utACK.
This will make next version 4.0.0 as it's a breaking change.

File: docs/examples/fetch-an-identity-from-its-name.md [hunk 1]
```diff
@@ -5,7 +5,9 @@ You will then be able to directly recover an identity from its names. See below:
 
 ```js
 const client = new Dash.Client({
-  mnemonic: ''// Your app mnemonic, which holds the app identity
+  wallet: {
+    mnemonic: '', // Your app mnemonic, which holds the app identity
```
Comment (thephez) [inline docs/examples/fetch-an-identity-from-its-name.md:9]: Remove obsolete "app id" reference
```suggestion
    mnemonic: '', // Your app mnemonic, which holds the identity
```

File: docs/examples/publishing-a-new-contract.md [hunk 2]
```diff
@@ -12,7 +12,9 @@ and [attached to a name](https://dashplatform.readme.io/docs/tutorial-register-a
 ```js
 const schema = {};// You JSON schema defining the app.
 const client = new Dash.Client({
-  mnemonic: ''// Your app mnemonic, which holds the app identity
+  wallet: {
+    mnemonic: '', // Your app mnemonic, which holds the app identity
```
Comment (thephez) [inline docs/examples/publishing-a-new-contract.md:16]:
```suggestion
    mnemonic: '', // Your app mnemonic, which holds the identity
```

File: docs/examples/publishing-a-new-contract.md [hunk 3]
```diff
@@ -40,7 +42,9 @@ const schema = {};// You JSON schema defining the app.
 
 // This is the name previously registered in DPNS.
 const client = new Dash.Client({
-  mnemonic: "",// Your app mnemonic, which holds the app identity
+  wallet: {
+    mnemonic: "", // Your app mnemonic, which holds the app identity
```
Comment (thephez) [inline docs/examples/publishing-a-new-contract.md:46]:
```suggestion
    mnemonic: "", // Your app mnemonic, which holds the identity
```

**Pull request merged**